tests/modules/apigee_mock_client/src/MockHttpClientFactory.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_mock_client;
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Http\ClientFactory;
+use Drupal\Core\Site\Settings;
+use Drupal\Core\State\StateInterface;
+use GuzzleHttp\Client;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+
+
+class MockHttpClientFactory extends ClientFactory {
+
+  /**
+   * The handler stack (retained for compatibility).
+   *
+   * @var \GuzzleHttp\HandlerStack
+   */
+  protected $stack;
+
+  /**
+   * The mock handler stack (Allows us to queue responses).
+   *
+   * @var \GuzzleHttp\HandlerStack
+   */
+  protected $mock_stack;
+
+  /**
+   * Whether or not integration is currently enabled.
+   *
+   * @var bool
+   */
+  protected $integration_enabled;
+
+  /**
+   * Constructs a new ClientFactory instance.
+   *
+   * @param \GuzzleHttp\HandlerStack $stack
+   *   The handler stack.
+   * @param \GuzzleHttp\Handler\MockHandler $mock_stack
+   * @param \Drupal\Core\State\StateInterface $state
+   */
+  public function __construct(HandlerStack $stack, MockHandler $mock_stack, StateInterface $state) {
+    $this->stack = $stack;
+    $this->mock_stack = $mock_stack;
+
+    // Check for the integration enabled environment variable.
+    if ($enabled = getenv('APIGEE_INTEGRATION_ENABLE')) {
```

Comment:
Use the name of the module in environment variables as prefix . `APIGEE_INTEGRATION_ENABLE ` is not unique enough.

Reply:
Maybe `APIGEE_MOCK_CLIENT_IS_MOCK_SERVER_ENABLED`?

Reply:
I went with `APIGEE_MOCK_CLIENT_MOCK_SERVER_ENABLED`, hope that's cool!